backend_new/src/services/application-csv-export.service.ts

```diff
@@ -0,0 +1,596 @@
+import fs from 'fs';
+import path from 'path';
+import { Inject, Injectable, Scope } from '@nestjs/common';
+import { REQUEST } from '@nestjs/core';
+import { Request as ExpressRequest } from 'express';
+import { view } from './application.service';
+import { PrismaService } from './prisma.service';
+import { MultiselectQuestionService } from './multiselect-question.service';
+import { ApplicationCsvQueryParams } from '../dtos/applications/application-csv-query-params.dto';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { UnitType } from '../dtos/unit-types/unit-type.dto';
+import { Address } from '../dtos/addresses/address.dto';
+import { ApplicationMultiselectQuestion } from 'src/dtos/applications/application-multiselect-question.dto';
+import MultiselectQuestion from '../dtos/multiselect-questions/multiselect-question.dto';
+import { ApplicationFlaggedSet } from '../dtos/application-flagged-sets/application-flagged-set.dto';
+import { UnitTypeService } from './unit-type.service';
+
+view.csv = {
+  ...view.details,
+  applicationFlaggedSet: true,
+  listings: false,
+};
+
+type CsvHeader = {
+  path: string;
+  label: string;
+  format?: (val: unknown) => unknown;
+};
+
+@Injectable({ scope: Scope.REQUEST })
+export class ApplicationCsvExporterService {
+  constructor(
+    @Inject(REQUEST) private req: ExpressRequest,
+    private prisma: PrismaService,
+    private multiselectQuestionService: MultiselectQuestionService,
+    private unitTypeService: UnitTypeService,
+  ) {}
+
+  /*
+   * Prepares an export and streams file
+   */
+  async export(queryParams: ApplicationCsvQueryParams): Promise<SuccessDTO> {
+    await this.authorizeCSVExport(this.req.user, queryParams.listingId);
+
+    if (queryParams.includeDemographics) {
+      view.csv.demographics = true;
+    }
+
+    const applications = await this.prisma.applications.findMany({
+      include: view.csv,
+      where: {
+        listingId: queryParams.listingId,
+      },
+    });
+
+    // get all multiselect questions for a listing to build csv headers
+    const multiSelectQuestions =
+      await this.multiselectQuestionService.findByListingId(
+        queryParams.listingId,
+      );
+
+    const csvHeaders = await this.getCsvHeaders(
+      multiSelectQuestions,
+      queryParams.includeDemographics,
+    );
+
+    const filePath = path.join(
+      path.resolve(process.cwd()),
+      `src/temp/listing-${queryParams.listingId}-applications.csv`,
```

Comment:
thoughts on adding the date to the file name string?

Reply:
good call, added a time string to the end of the file